benchmarks/linear-elastic-plate-with-hole/workflow_creation_guide.md

```diff
@@ -0,0 +1,930 @@
+    # Discover parameter files
+    files = list(Path(".").glob("parameters_*.json"))
+    
+    # Extract and validate configurations
+    configurations = {}
+    for file in files:
+        with open(file, 'r') as f:
+            data = json.load(f)
+        config_name = data.get("configuration", file.stem.split("_")[1])
+        configurations[file] = config_name
+    
+    # Check for duplicates
+    config_values = list(configurations.values())
+    duplicates = set([x for x in config_values if config_values.count(x) > 1])
+    if duplicates:
+        raise ValueError(f"Duplicate configurations: {duplicates}")
+    
+    # Create workflow configuration
+    workflow_config = {
+        "configuration_to_parameter_file": {v: str(k) for k, v in configurations.items()},
+        "configurations": list(configurations.values()),
+        "tools": ["fenics", "kratos"],  # Update with your tools
+        "benchmark": "your-benchmark-name"
+    }
+    
+    # Save configuration
+    with open("workflow_config.json", "w") as f:
+        json.dump(workflow_config, f, indent=4)
+    
+    return workflow_config
+
+if __name__ == "__main__":
+    generate_workflow_config()
+```
+
+### Step 3: Implement Mesh Generation
+
+Create `create_mesh.py` with standardized interface:
+
+```python
+import json
+import gmsh
+from argparse import ArgumentParser
+from pint import UnitRegistry
+
+def create_mesh(parameter_file: str, mesh_file: str):
+    ureg = UnitRegistry()
+    
+    # Load parameters
+    with open(parameter_file) as f:
+        parameters = json.load(f)
+    
+    configuration = parameters["configuration"]
+    
+    # Extract geometry parameters with unit conversion
+    length = ureg.Quantity(
+        parameters["length"]["value"], 
+        parameters["length"]["unit"]
+    ).to_base_units().magnitude
+    
+    radius = ureg.Quantity(
+        parameters["radius"]["value"], 
+        parameters["radius"]["unit"]
+    ).to_base_units().magnitude
+    
+    element_size = ureg.Quantity(
+        parameters["element-size"]["value"], 
+        parameters["element-size"]["unit"]
+    ).to_base_units().magnitude
+    
+    # Initialize gmsh and create mesh
+    gmsh.initialize()
+    gmsh.model.add(configuration)
+    gmsh.option.setNumber("Mesh.CharacteristicLengthMin", element_size)
+    
+    # Create geometry (implementation specific to your problem)
+    # ... geometry creation code ...
+    
+    # Generate mesh
+    gmsh.model.mesh.generate(2)
+    gmsh.write(mesh_file)
+    gmsh.finalize()
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="Generate mesh for benchmark problem")
+    parser.add_argument("--input_parameter_file", required=True)
+    parser.add_argument("--output_mesh_file", required=True)
+    args = parser.parse_args()
+    
+    create_mesh(args.input_parameter_file, args.output_mesh_file)
+```
+
+### Step 4: Create Snakemake Workflow System
+
+#### Main Snakefile
+
+```snakemake
```

Comment:
here link the snakemake docu and the example file and highlight some important featues that are used (e.g. wildcards) and maybe even link those features directly in the snakemake docu, similarly document the approach in the snakefile